.github/workflows/ui.yml

```diff
@@ -0,0 +1,63 @@
+name: UI Tests
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+
+jobs:
+  run:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        python: [3.6, 3.7, 3.8, 3.9]
+
+    name: Python ${{ matrix.python }}
+
+    defaults:
+      run:
+        shell: bash -l {0}
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Python ${{ matrix.python }}
+        uses: actions/setup-python@v2.3.1
+        with:
+          python-version: ${{ matrix.python }}
+
+      - name: Setup poetry
+        uses: abatilo/actions-poetry@v2.1.4
+        with:
+          poetry-version: 1.1.6
+
+      - name: Configure poetry
+        run: poetry config virtualenvs.in-project true
+
+      - name: Install Python dependencies
+        run: poetry install
+      - run: poetry install --extras "jupyterlab"
+
+      - name: Setup Node
+        uses: actions/setup-node@v2.5.1
+
+      - name: Run tests
+        run: |
+          yarn install
+          yarn run build
+          poetry run jupyter labextension install .
+          export PATH="$(poetry env info -p)/bin:$PATH"
+          cd ui-tests/
+          yarn
```

Comment:
Please move the dependencies for the ui tests into the top-level package (as dev dependencies) so we don't need a second package.json and lock file. We can probably also move the script commands up to the top level.

Reply:
We have added an end-to-end integration test of ipyvega using Galata. We have used the mechanism described by Galata and used by the ipywidgets (see https://github.com/jupyter-widgets/ipywidgets/tree/master/ui-tests).
We test that running ipyvega on a specification produces exactly the same output figure when run on the notebook.
We don't know of any example that merges the end-to-end integration test with the main package.json, so it would be easier for us to remove altogether the end-to-end test. We thought it was a useful addition to make sure the visualizations would not change inadvertently, but it was not something we agreed on adding so removing it is fine.

Reply:
Ah, I see that ipywidget uses the same approach. Let's leave it as is then.